app/services/video-encoding-optimizations/index.ts

```diff
@@ -0,0 +1,110 @@
+import { Service } from 'services/service';
+import { SettingsService, ISettingsSubCategory } from 'services/settings';
+import { StreamingService } from 'services/streaming';
+import { Inject } from '../../util/injector';
+import { IProfile, IEncoderPreset, getPreset } from './definitions';
+
+const cloneDeep = require('lodash.clonedeep');
+
+let previousSettings: any;
+let isUsingEncodingOptimizations: boolean = false;
+
+export class VideoEncodingOptimizationService extends Service {
+  @Inject() settingsService: SettingsService;
+  @Inject() streamingService: StreamingService;
+
+  init() {
+    this.streamingService.streamingStatusChange.subscribe(
+        status => {
+          if (!status) {
+            isUsingEncodingOptimizations = false;
```

Comment:
You should probably check that they are using encoding optimizations here, otherwise you are going to `restorePreviousValues` if they stop streaming and aren't using optimized encoders.

Reply:
Ooops ahah

Reply:
Done.